mario/pipelines/app_barsdetection.py

```diff
@@ -0,0 +1,128 @@
+        from requests import post
+        from sonyci import SonyCi
+        from sqlmodel import Session, select
+
+        # get SonyCi Asset ID
+        with Session(engine) as db:
+            media_file = db.exec(
+                select(MediaFile).where(MediaFile.guid == self.guid)
+            ).one()
+            self.asset_id = media_file.assets[0].id
+
+        assert self.asset_id, f'No asset found for {self.guid}'
+        print(f'Found asset {self.asset_id}')
+        self.filename = join('/m', self.guid + '.mp4')
+
+        # get mmif
+        self.input_mmif = self.mmif
+        if not self.mmif:
+            print('No mmif provided, downloading from clams')
+            self.input_mmif = post(
+                'http://fastclam/source',
+                json={'files': ['video:' + self.filename]},
+            ).json()
+        print('Got mmif')
+        print(self.input_mmif)
+        # Download the media file
+        # run(f'ci download {self.asset_id} -o {join("/m", self.guid)}')
+        ci = SonyCi(**SonyCi.from_env())
+        self.asset = ci.get(f'assets/{self.asset_id}')
+
+        url = self.asset['proxyUrl']
+        print('Downloading file')
+        urlretrieve(url, self.filename)
+
+        print('Downloaded file')
+        run(['ls', '-al', '/m'])
+
+        self.next(self.barsdetection)
+
+    @kubernetes(image='ghcr.io/wgbh-mla/mario:pr-4')
+    @step
+    def barsdetection(self):
+        """Run the mmif through app-barsdetection"""
+        from requests import post
+
+        self.response = post('http://app-barsdetection/', json=self.input_mmif)
+        if self.response.status_code != 200:
+            print(self.response.content)
+            from mario.utils import CLAMSAppError
+
+            raise CLAMSAppError(
+                f'app-barsdetection failed: {self.response.status_code} - {self.response.content}'
+            )
+        self.output_mmif = self.response.json()
+        print('Response from app-barsdetection:')
+        print(self.output_mmif)
+
+        self.next(self.end)
+
+    @kubernetes(
+        image='ghcr.io/wgbh-mla/mario:main',
+        persistent_volume_claims={
+            'media-pvc': '/m',
+        },
+    )
+    @step
+    def end(self):
+        """Report results and cleanup"""
+        from glob import glob
+        from json import dumps
+        from os import remove
+        from os.path import join
+        from subprocess import run
+
+        from boto3 import client
+
+        mmif_filename = join('/m', self.guid + '.mmif')
+        s3_path = f'{self.guid}/app-barsdetection/{self.guid}.mmif'
+
+        with open(mmif_filename, 'w') as f:
+            f.write(dumps(self.output_mmif))
+
+        run(['ls', '-al', '/m'])
+
+        # Upload transcript to aws
+        print(f'Uploading {mmif_filename} to {s3_path}')
+        client = client('s3')
+        client.upload_file(
+            mmif_filename,
+            'clams-transcripts',
```

Comment:
different bucket prefix?

Reply:
I'll create an mmif bucket now and close out https://github.com/WGBH-MLA/chowda-deploy/issues/4 as well!